src/semgrep_agent/meta.py

```diff
@@ -55,6 +56,10 @@ def repo_url(self) -> Optional[str]:
     def commit_sha(self) -> Optional[str]:
         return self.repo.head.commit.hexsha  # type: ignore
 
+    @cachedproperty
+    def head_ref(self) -> Optional[str]:
```

Comment:
Is it fine that this is set to None for non Github projects?

Reply:
Yes, if this is `None`, the entire work here is a no-op.